src/UI/Utils/ErrorBoundary.test.tsx

```diff
@@ -12,9 +13,11 @@ test("GIVEN ErrorBoundary WHEN no error is thrown THEN error page should NOT be
   const explode = false;
 
   const component = (
-    <ErrorBoundary>
-      <span data-testid="bomb">{explode ? <Bomb /> : "normal text"}</span>
-    </ErrorBoundary>
+    <MemoryRouter>
```

Comment:
The ErrorBoundary should by design be encapsulating the entire application. Having it wrapped in a MemoryRouter defeats the purpose of the element, even if it is in a test case. You shouldn't have anything around it.

Reply:
We are using navigate in the errorBoundary to redirect, putting Router inside cause error, so only thing I can do is to change into `window.location.replace()`

Reply:
The ErrorBoundary is for unhandled errors. It gets displayed only in those cases, and shouldn't be responsible for redirecting the user. To me, it means the logic has flaws if you end up here. This isn't resolving the issue, but hacking around.